smt.go

```diff
@@ -319,7 +323,7 @@ func (smt *SparseMerkleTree) updateWithSideNodes(path []byte, value []byte, side
 // Returns an array of sibling nodes, the leaf hash found at that path and the
 // leaf data. If the leaf is a placeholder, the leaf data is nil.
 func (smt *SparseMerkleTree) sideNodesForRoot(path []byte, root []byte) ([][]byte, []byte, []byte, error) {
-	sideNodes := make([][]byte, smt.depth())
+	var sideNodes [][]byte
```

Comment:
Allocating required size was a good step. As _slice tricks_ are used below, we should at least set size to `0` and capacity to `smt.depth()`, with `sideNodes := make([][]byte, 0, smt.depth())`

Reply:
I wasn't actually aware of this syntax, but was looking for something like that! Good fix.